pkgs/applications/misc/when/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, perl
+, installShellFiles
+, fetchFromBitbucket
+}:
+
+stdenv.mkDerivation {
+  pname = "when";
+  version = "1.1.46";
+
+  src = fetchFromBitbucket {
+    owner = "ben-crowell";
+    repo = "when";
+    rev = "a57ccbdf1d30e28a43a4bc5c68c37c171bb649ab";
+    hash = "sha256-uNIU4tsoJBBPj+2rYVvXl+D5f/GB80ox6s2MgxjuJa4=";
+  };
+
+  buildInputs = [ perl ];
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install -m 755 when $out/bin/
```

Comment:
I just put it together that this is a Perl script. You need to `patchShebangs` for it to work correctly.

Reply:
This should be done automatically
https://nixos.org/manual/nixpkgs/stable/#ssec-fixup-phase

Reply:
Indeed :+1:.